src/main/java/de/tum/cit/aet/artemis/lecture/domain/LectureUnit.java

```diff
@@ -159,6 +172,20 @@ public Optional<ZonedDateTime> getCompletionDate(User user) {
         return getCompletedUsers().stream().filter(completion -> completion.getUser().getId().equals(user.getId())).map(LectureUnitCompletion::getCompletedAt).findFirst();
     }
 
+    @PrePersist
+    @PreUpdate
+    private void updateLectureUnitOrder() {
+        if (lecture == null) {
+            return; // should not really happen due to nullable=false, but safe
+        }
+        lecture.updateLectureUnitOrder();
+    }
```

Comment:
Do we really need this here? Wouldn't it be better to only keep that in the lecture?

I am not sure about the internals, but couldn't this lead to issues with the `PrePersist` in lecture?

1. saveAndFlush(lecture) -> @PreUpdate on Lecture -> updateLectureUnitOrder()
2. @PrePersist on LectureUnit -> lecture.updateLectureUnitOrder() _(I am not sure how it works here, but if it goes back to lecture due to the changes again we would have a circle)_
3. @PreUpdate on Lecture -> updateLectureUnitOrder()